executor/src/lib.rs

```diff
@@ -169,3 +200,73 @@ impl Executor {
         ])
     }
 }
+
+fn replace_distant_by_localhost(target: &mut Multiaddr, hostname_pattern: &Protocol) {
```

Comment:
Maybe logging the successful replace can give us in staging/devnet a quick feedback whether we have successfully overwritten the address